tests/bsim/bluetooth/audio/src/bap_broadcast_sink_test.c

```diff
@@ -1027,7 +1045,38 @@ static void test_sink_encrypted(void)
 		k_sem_take(&sem_stopped, K_FOREVER);
 	}
 
-	PASS("Broadcast sink passed\n");
+	PASS("Broadcast sink encrypted passed\n");
+}
+
+static void test_sink_encrypted_incorrect_code(void)
+{
+	int err;
+
+	err = init();
+	if (err) {
+		FAIL("Init failed (err %d)\n", err);
+		return;
+	}
+
+	test_scan_and_pa_sync();
+
+	test_broadcast_sink_create();
+
+	printk("Broadcast source PA synced, waiting for BASE\n");
+	WAIT_FOR_FLAG(flag_base_received);
+	printk("BASE received\n");
+
+	printk("Waiting for BIG syncable\n");
+	WAIT_FOR_FLAG(flag_syncable);
+
+	test_broadcast_sync(INCORRECT_BROADCAST_CODE);
+	/* Wait for MIC failure */
+	WAIT_FOR_FLAG(flag_big_sync_mic_failure);
```

Comment:
We're inconsistent with the use of FLAGS; for some tests we have an init function where we initialise the flags, sometimes we initialise them in the test-function itself, and sometimes we do not initialise them. Probably out of scope for this PR, but I think it is better to initialise them.

Reply:
You mean the `UNSET_FLAG`? 

It's not really needed unless the flag is triggered multiple times in a test, but we could be more consistent.

Reply:
Yes, that's what I mean.